ompi/mpi/c/sendrecv.c

```diff
@@ -92,21 +92,27 @@ int MPI_Sendrecv(const void *sendbuf, int sendcount, MPI_Datatype sendtype,
     if (dest != MPI_PROC_NULL) { /* send */
         rc = MCA_PML_CALL(send(sendbuf, sendcount, sendtype, dest,
                                sendtag, MCA_PML_BASE_SEND_STANDARD, comm));
+        if (OPAL_UNLIKELY(MPI_SUCCESS != rc)) {
+            rcs = rc;
 #if OPAL_ENABLE_FT_MPI
-        /* If ULFM is enabled we need to wait for the posted receive to
-         * complete, hence we cannot return here */
-        rcs = rc;
-#else
-        OMPI_ERRHANDLER_CHECK(rc, comm, rc, FUNC_NAME);
-#endif  /* OPAL_ENABLE_FT_MPI */
+            /* If this is a PROC_FAILED error, we still need to proceed with
+             * the receive, so that we do not propagate errors to the sender in
+             * the case src != dst, and only dst is dead. In this case the 
+             * recv is garanteed to complete (either in error if the source is 
+             * dead, or successfully if the source is live). */
+            if (OPAL_UNLIKELY(MPI_ERR_PROC_FAILED != rc))
+            /* if intentionally spills outside ifdef */
+#endif
+            ompi_request_cancel(req);
+        }
     }
 
     if (source != MPI_PROC_NULL) { /* wait for recv */
         rc = ompi_request_wait(&req, status);
 #if OPAL_ENABLE_FT_MPI
```

Comment:
don't we need a matching endif ?

Reply:
line 115 below. It does not have the comment to indicate what it is matched with, but it's there.

Reply:
hmmph.  i do need new glasses